app/invocation/cache_requests_card.tsx

```diff
@@ -368,7 +371,62 @@ export default class CacheRequestsCardComponent extends React.Component<CacheReq
     ));
   }
 
+  private getCacheMetadata(scorecardResult: cache.ScoreCard.IResult) {
+    const digest = scorecardResult.digest;
+    const remoteInstanceName = this.props.model.getRemoteInstanceName();
+    rpc_service.service
+      .getCacheMetadata(
+        new cache.GetCacheMetadataRequest({
+          resourceName: {
+            digest: digest,
+            cacheType: scorecardResult.cacheType,
+            instanceName: remoteInstanceName,
+          },
+        })
+      )
+      .then((response) => {
+        this.setState(function (prevState, props) {
+          prevState.digestToCacheMetadata.set(digest.hash, response);
+          return {
+            digestToCacheMetadata: prevState.digestToCacheMetadata,
+          };
+        });
```

Comment:
optional nit: for this sort of state update (where you aren't changing a state key but still want the component to re-render), I usually prefer to mutate the state then call `forceUpdate()` to inform React about the state update. I feel like the alternative of calling `setState()` and setting a state key equal to itself is less clear.

```suggestion
        this.state.digestToCacheMetadata.set(digest.hash, response);
        this.forceUpdate();
```

Reply:
Done